packages/bun-usockets/src/context.c

```diff
@@ -63,16 +62,19 @@ void us_socket_context_close(int ssl, struct us_socket_context_t *context) {
     /* Begin by closing all listen sockets */
     struct us_listen_socket_t *ls = context->head_listen_sockets;
     while (ls) {
-        struct us_listen_socket_t *nextLS = (struct us_listen_socket_t *) ls->s.next;
+        struct us_listen_socket_t *nextLS = (struct us_listen_socket_t *) ls->s.next;\
         us_listen_socket_close(ssl, ls);
+        
         ls = nextLS;
     }
 
     /* Then close all regular sockets */
     struct us_socket_t *s = context->head_sockets;
     while (s) {
         struct us_socket_t *nextS = s->next;
-        us_socket_close(ssl, s, LIBUS_SOCKET_CLOSE_CODE_CLEAN_SHUTDOWN, 0);
```

Comment:
We only need this check when the context is an SSL context, right?

Reply:
we need to check internally anyway, so I removed from this place, we have to refactor a lot more before making this an assertion